Library/Homebrew/cmd/fetch.rb

```diff
@@ -87,52 +88,92 @@ def self.fetch
       args.named.to_formulae_and_casks
     end.uniq
 
+    os_arch_combinations = args.os_arch_combinations
+
     puts "Fetching: #{bucket * ", "}" if bucket.size > 1
     bucket.each do |formula_or_cask|
       case formula_or_cask
       when Formula
-        f = formula_or_cask
-
-        f.print_tap_action verb: "Fetching"
-
-        fetched_bottle = false
-        if fetch_bottle?(f, force_bottle: args.force_bottle?, bottle_tag: args.bottle_tag&.to_sym,
-                         build_from_source_formulae: args.build_from_source_formulae)
-          begin
-            f.clear_cache if args.force?
-            f.fetch_bottle_tab
-            fetch_formula(f.bottle_for_tag(args.bottle_tag&.to_sym), args: args)
-          rescue Interrupt
-            raise
-          rescue => e
-            raise if Homebrew::EnvConfig.developer?
-
-            fetched_bottle = false
-            onoe e.message
-            opoo "Bottle fetch failed, fetching the source instead."
-          else
-            fetched_bottle = true
-          end
-        end
+        formula = T.cast(formula_or_cask, Formula)
+        ref = formula.loaded_from_api? ? formula.full_name : formula.path
```

Comment:
This is used in multiple places; maybe worth a dedicated method on `Formula`?

Reply:
I guess we should have something like `Formula#refresh`. I think for casks we actually already have `Cask#refresh`.

Reply:
That's significantly harder for formulae though, given the class is already evaluated for a specific system if the formula instance exists.

Reply:
Fine with me to leave as-is.